File: Modulo09/TimeJogador-Java/Jogador.java
public class Jogador {
    private String nome;
    private int idade;
    private String posicao;
    private Time time;

    public Jogador(String nome, int idade, String posicao) {
        this.nome = nome;
        this.idade = idade;
        this.posicao = posicao;
        System.out.println("Criando jogador " + this.nome);
    }

    public String getNome() {
        return this.nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public int getIdade() {
        return this.idade;
    }

    public void setIdade(int idade) {
        this.idade = idade;
    }

    public String getPosicao() {
        return this.posicao;
    }

    public void setPosicao(String posicao) {
        this.posicao = posicao;
    }

    public Time getTime() {
        return this.time;
    }

    public void setTime(Time time) {
        this.time = time;
    }

    public void imprime() {
        System.out.print(nome + ", " + posicao + ", " + idade + " anos");
        if (time != null) {
            System.out.print(" (Joga no " + time.getNome() + ")");
        } else {
            System.out.print(" (Aposentado/Desempregado)");
        }
        System.out.println();
    }
}

File: Modulo06/TimeJogador-v2/Jogador.cpp
#include "Jogador.h"
#include "Time.h"

#include <iostream>
using namespace std;

Jogador::Jogador(string nome, int idade, string posicao) :
nome(nome), idade(idade), posicao(posicao) {
    cout << "Criando jogador " << nome << endl;
}

Jogador::~Jogador() {
    cout << "Destruindo jogador " << nome << endl;
}

string Jogador::getNome() const {
    return nome;
}

void Jogador::setNome(string nome) {
    this->nome = nome;
}

int Jogador::getIdade() const {
    return idade;
}

void Jogador::setIdade(int idade) {
    this->idade = idade;
}

string Jogador::getPosicao() const {
    return posicao;
}

void Jogador::setPosicao(string posicao) {
    this->posicao = posicao;
}

Time* Jogador::getTime() const {
    return time;
}

void Jogador::setTime(Time* time) {
    this->time = time;
}

void Jogador::imprime() const {
    cout << nome << ", " << posicao << ", " << idade << " anos";
    if (time != NULL) {
        cout << " (Joga no " << time->getNome() << ")";
    } else {
        cout << " (Aposentado/Desempregado)";
    }
    cout << endl;
}